docs/site/LB3-vs-LB4-booting.md

```diff
@@ -0,0 +1,128 @@
+---
+lang: en
+title: 'Differences in LoopBack 3 and LoopBack 4 booting process'
+keywords: LoopBack 4.0, LoopBack 4, LoopBack 3.0, LoopBack 3
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/LB3-vs-LB4-booting.html
+---
+
+## Differences between LoopBack 3 and LoopBack 4 booting process
+
+The booting process in LoopBack 3 is handled by
+[loopback-boot](https://github.com/strongloop/loopback-boot),
+which performs the following tasks:
+
+### In LoopBack 3
+
+#### 1. Configuration of application settings
+
+It determines the execution environment, the application host, the application
+port, the application root directory and other artifact directories.
+
+Then it configures the LoopBack Application object from the `config.json` file
+in the application root directory.
+
+#### 2. Configuration of datasources
+
+Datasources are determined from application options or datasource files. They
+are then attached to the LoopBack Application object.
+
+#### 3. Definition of models
+
+Models are determined from application options or model files,
+[mixins](https://loopback.io/doc/en/lb3/Defining-mixins.html) are applied,
+and they are attached to their respective datasources. They are then attached
+to the LoopBack Application object.
```

Comment:
FYI: In the first step, we _define_ all models. In the second step, we _attach_ the models to datasources and the application.

The list of models defined may be longer than the list of models attached:
1. loopback-boot looks at `server/model-config` to see which files should be added to the app.
2. Then it examines inheritance and mixin configurations, to determine dependencies (which models to add to the list)
3. Models are sorted topologically based on inheritance, so that we define parent models before children.
4. All models from the list above are defined via model registry.
5. Only models specified in `server/model-config` are added to the app and attached to a datasource (unless `dataSource` flag is set to `null`).

loopback-boot also supports universal mode, where the entire loopback application is bundled for browser. In that case, the boot process is split in two parts:
1. The first step finds out which models (and their parents, etc.) are needed by the app. It creates "instructions" for booting an app, these instructions are compiled into the browser bundle together with model source files (`common/models/{model-name}.json`).
2. The second step is performed in the browser, it processes the instructions from the first step.

Reply:
Maybe not worth getting into all those details for something that we are moving away from. Especially since the corresponding step in LB4 is an `import`.